common/OpTestHMC.py

```diff
@@ -0,0 +1,348 @@
+            log.info('LPAR Already powered on!')
+            return
+        lpar_name = self.lpar_name
+        if vios:
+            lpar_name = self.lpar_vios
+        cmd = "chsysstate -m %s -r lpar -n %s -o on" % (self.system, lpar_name)
+
+        if not vios:
+            if self.lpar_prof:
+                cmd = "%s -f %s" % (cmd, self.lpar_prof)
+
+        self.wait_lpar_state(OpHmcState.NOT_ACTIVE, vios=vios)
+        self.run_command(cmd)
+        self.wait_lpar_state(vios=vios)
+        if runtime:
+            self.wait_login_prompt(self.get_console_prompt())
+            self.close_console(self.lpar_con)
+
+    def get_lpar_state(self, vios=False):
+        lpar_name = self.lpar_name
+        if vios:
+            lpar_name = self.lpar_vios
+        state = self.run_command(
+            'lssyscfg -m %s -r lpar --filter lpar_names=%s -F state' % (self.system, lpar_name))
+        return state[-1]
+
+    def get_system_state(self):
+        state = self.run_command(
+            'lssyscfg -m %s -r sys -F state' % self.system)
+        return state[-1]
+
+    def wait_lpar_state(self, exp_state=OpHmcState.RUNNING, vios=False, timeout=WAITTIME):
+        state = self.get_lpar_state(vios)
+        count = 0
+        while state != exp_state:
+            state = self.get_lpar_state(vios)
+            log.info("Current state: %s" % state)
+            time.sleep(timeout)
+            count = 1
+            if count > 120:
+                raise OpTestError("Time exceeded for reaching %s" % exp_state)
+
+    def wait_system_state(self, exp_state=OpManagedState.OPERATING, timeout=WAITTIME):
+        state = self.get_system_state()
+        count = 0
+        while state != exp_state:
+            state = self.get_system_state()
+            log.info("Current state: %s" % state)
+            time.sleep(timeout)
+            count = 1
+            if count > 60:
+                raise OpTestError("Time exceeded for reaching %s" % exp_state)
+
+    def console_run_command(self, console, cmd, timeout=60):
+        if console is None:
+            console = self.get_console()
+
+        try:
+            console.send(cmd)
+            console.send('\r')
+            output = console.before
+            console.sendline("echo $?")
+            console.send('\r')
+            # Time to accumulate both outputs
+            time.sleep(STALLTIME)
+            rc = console.expect(["\[console-pexpect\]#$"], timeout=timeout)
+            if rc != 0:
+                exitcode = int(console.before)
+                log.debug("# LAST COMMAND EXIT CODE %d (%s)" % (exitcode, repr(console.before)))
+        except pexpect.TIMEOUT as e:
+            log.debug(e)
+            log.debug("# TIMEOUT waiting for command to finish.")
+            log.debug("# Attempting to control-c")
+            try:
+                console.sendcontrol('c')
+                rc = console.expect(["\[console-pexpect\]#$"], 10)
```

Comment:
Suggest using the utility to build the prompt to provide a system wide common abstraction.